### PR TITLE
Binding ref to VueComponent does not working

### PR DESCRIPTION
**Pull request by likeswinds**
#320

### CODE REVIEW

File: packages/core/useMouseInElement/index.ts [hunk 1]
```diff
@@ -39,7 +39,7 @@ export function useMouseInElement(
     stop = watch(
       [targetRef, x, y],
       () => {
-        const el = targetRef.value
+        const el = targetRef.value.$el || targetRef.value
```
Comment (antfu) [inline packages/core/useMouseInElement/index.ts:42]:
```suggestion
        const el = targetRef.value?.$el || targetRef.value
```

File: packages/core/onClickOutside/index.ts [hunk 2]
```diff
@@ -31,7 +31,8 @@ export function onClickOutside(
       return
 
     const elements = event.composedPath()
-    if (targetRef.value === event.target || elements.includes(targetRef.value!))
+    const el = targetRef.value.$el || targetRef.value
```
Comment (antfu) [inline packages/core/onClickOutside/index.ts:34]:
```suggestion
    const el = targetRef.value?.$el || targetRef.value
```

**Comment by likeswinds** [comment]
image
I can't do it this way @antfu 
 ``` 
const el = (targetRef.value as any).$el || targetRef.value
```

**Comment by likeswinds** [comment]
A similar problem exists in all of these functions，such as `useElementVisibility`,`useElementBounding`,`useElementSize`,`useMutationObserver`,`useCssVar` and so on 
Or I should program it this way 
``` vue
const modal = templateRef('modal')
onMounted(() => {
  onClickOutside(modal.value.$el, event => {
    console.log(event)
  })
})
```
Here are some code snippets
```
// useElementVisibility , 32 (row)
const rect = element.value.getBoundingClientRect()  // element.value.$el.getBoundingClientRect() 
```
```
// useResizeObserver, 69(row)
observer!.observe(newValue, observerOptions)
```

**Comment by antfu** [comment]
@likeswinds Thanks for the feedback, now it's supported since v4.1

**Pull request closed without merging**